libs/hdf-converters/src/xccdf-results-mapper.ts

```diff
@@ -23,79 +23,70 @@ const IMPACT_MAPPING: Map<string, number> = new Map([
 
 const CCI_NIST_MAPPING = new CciNistMapping();
 
-const RULE_RESULT_PATHS = ['cdf:rule-result', 'rule-result'];
-
-let idTracker = '';
-let valueIdTracker: string | undefined = undefined;
+function asArray<T>(arg: T | T[]): T[] {
+  if (Array.isArray(arg)) {
+    return arg;
+  } else if (arg === undefined || arg === null) {
+    return [];
+  } else {
+    return [arg];
+  }
+}
 
-function getRuleResultItem(
-  testResult: Record<string, unknown>,
-  pathRuleResultPossibilities: string[],
-  pathIdRefPossibilities: string[] = ['idref'],
-  pathItemPossibilities: string[] | undefined = undefined
-): unknown {
-  for (const pathRuleResult of pathRuleResultPossibilities) {
-    const ruleResult: Record<string, unknown>[] | undefined = _.get(
-      testResult,
-      pathRuleResult
-    ) as Record<string, unknown>[] | undefined;
-    if (ruleResult === undefined) {
+function getRuleResult(
+  ruleId: string,
+  benchmark: Record<string, unknown>
+): Record<string, unknown> | undefined {
+  for (const pathRuleResult of [
+    'cdf:TestResult.cdf:rule-result',
```

Comment:
After we'd written this, we saw that there might be ways to make this easier.

https://github.com/mitre/heimdall2/issues/3084

The blocker for this is updating the version of the fast-xml-parser, which I could look into hopefully soon.

Reply:
That got merged

Reply:
I've updated this MR and this change certainly does make the task much easier and less error prone!